lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -335,8 +353,19 @@ def buildUI(self):
             if schemaType.pythonClass:
                 attrsToAdd = schemaType.pythonClass.GetSchemaAttributeNames(False)
 
+                # remove all array attriutes from the attrsToAdd list
+                if not self.showArrayAttributes:
+                    attrsToAdd = [x for x in attrsToAdd if not self.isArrayAttribute(x)]
```

Comment:
Typo in your comment "attributes".
Here you are calling isArrayAttribute() again on ones that you already called on (from the __init__). We should only call this method once on each attribute.

Reply:
this is to avoid calling addControl on the attribute. The one in init is called for suppress the attribute. Otherwise the attribute will be in the "Extra" section

Reply:
Understood, but calling isArrayAttribute twice is a little overkill. How about if where you first call it (to suppress) you maintain a list. Then here or maybe even better in the addControls you can check the list.